pandas/tests/groupby/test_groupby.py

```diff
@@ -2907,3 +2907,30 @@ def test_groupby_cumsum_mask(any_numeric_ea_dtype, skipna, val):
         dtype=any_numeric_ea_dtype,
     )
     tm.assert_frame_equal(result, expected)
+
+def test_groupby_index_name_in_index_content():
+    # GH 48567
+    series1 = Series(data=[1.0, 2.0, 3.0, 4.0, 5.0], name='values',
+                     index=Index(['foo', 'foo', 'bar', 'baz', 'blah'],
+                                 name='blah'))
+    result1 = series1.groupby('blah').sum()
+    expected1 = Series(data=[3.0, 4.0, 5.0, 3.0], name='values',
+                       index=Index(['bar', 'baz', 'blah', 'foo'], name='blah'))
+    tm.assert_series_equal(result1, expected1)
+
+    series2 = Series(data=[1.0, 2.0, 3.0, 4.0, 5.0, 6.0], name='values',
```

Comment:
Could you use `pytest.mark.parameterize` to simplify these similar tests?

Also looks like theres formatting issues here: https://github.com/pandas-dev/pandas/actions/runs/3122348100/jobs/5064232051

Reply:
I will try to simplify the tests using pytest.mark.parameterize.

I am not able to use pre-commit (there is a known bug with posix_prefix, posix_local and virtualenv) on my system. I am still searching for a solution. Is there another way to check formatting?

Reply:
pre-commit runs black and flake8 so if you're able to run those manually it should be comparable